### PR TITLE
Remove Rails edge for Ruby 2.7

### PR DESCRIPTION
**Pull request by marcin-drozd**
Fixes https://github.com/Shopify/payments-platform/issues/4567

Add Ruby 2.7 to exclude Rails edge similarly as [it was done in the past for Ruby 2.6](https://github.com/activemerchant/payment_icons/pull/404) to see if this resolves CI issues we've been getting ([example failure](https://github.com/activemerchant/payment_icons/pull/1120))

```ruby
> bundle install
/opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle config --local path /home/runner/work/payment_icons/payment_icons/vendor/bundle
/opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle lock
Fetching https://github.com/rails/rails.git
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Could not find compatible versions

Because every version of rails depends on Ruby >= 3.1.0
  and rails-edge.gemfile depends on rails >= 0,
  Ruby >= 3.1.0 is required.
So, because current Ruby version is = 2.7.8,
  version solving has failed.
```

### CODE REVIEW

**Comment by larouxn** [comment]
Alternatively, we could remove support for Ruby 2.6 and 2.7. Both are EOL already anyway.
image

( will leave the decision up to the current maintainers :pray: )

**Comment by marcin-drozd** [comment]
Removing the support for those versions makes sense 👍  I think we should definitely look into this and see how things are used, but I'd like to merge this just to unblock partners for now if that makes sense.

**Comment by larouxn** [comment]
> Removing the support for those versions makes sense 👍 I think we should definitely look into this and see how things are used, but I'd like to merge this just to unblock partners for now if that makes sense.

Makes sense to me. :+1:

**Review by larouxn — APPROVED** [review]

**Review by adeniyiao — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**
